src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ExperimentalReadThreadingGraph.java

```diff
@@ -0,0 +1,796 @@
+     *
+     * If a node has an in-degree > 1, then it has a junction tree added to the previous node.
+     *
+     * @param prevVertex Current vertex to extend off of
+     * @param sequence Sequence of kmers to extend
+     * @param kmerStart index of where the current kmer starts
+     * @return The next vertex in the sequence. Null if the corresponding edge doesn't exist.
+     */
+    protected MultiDeBruijnVertex extendJunctionThreadingByOne(final MultiDeBruijnVertex prevVertex, final byte[] sequence, final int kmerStart, List<ThreadingNode> nodesToExtend) {
+        final Set<MultiSampleEdge> outgoingEdges = outgoingEdgesOf(prevVertex);
+
+        final int nextPos = kmerStart + kmerSize - 1;
+        for (final MultiSampleEdge outgoingEdge : outgoingEdges) {
+            final MultiDeBruijnVertex target = getEdgeTarget(outgoingEdge);
+            if (target.getSuffix() == sequence[nextPos]) {
+                // Only want to create a new tree if we walk into a node with
+                // NOTE: we do this deciding on our path
+                if (vertexWarrantsJunctionTree(prevVertex)) {
+                    nodesToExtend.add(readThreadingJunctionTrees.computeIfAbsent(prevVertex, k -> new ThreadingTree(prevVertex)).getAndIncrementRootNode());
+                }
+
+                // If this node has an out-degree > 1, add the edge we took to existing trees
+                if (outgoingEdges.size() != 1) {
+                    // TODO, make an object to encapsulate this operation better
+                    addEdgeToJunctionTreeNodes(nodesToExtend, outgoingEdge);
+                }
+
+                return target;
+            }
+        }
+        return null;
+    }
+
+    // Helper method that adds a single edge to all of the nodes in nodesToExtend.
+    private static void addEdgeToJunctionTreeNodes(List<ThreadingNode> nodesToExtend, MultiSampleEdge outgoingEdge) {
+        List<ThreadingNode> newNodes = nodesToExtend.stream().map(n -> n.addEdge(outgoingEdge)).collect(Collectors.toList());
+        nodesToExtend.clear();
+        nodesToExtend.addAll(newNodes);
+    }
+
+    // Helper method used to determine whether a vertex meets the criteria to hold a junction tree
+    // The current criteria, if any outgoing edge from a particular vertex leads to a vertex with inDegree > 1, then it warrants a tree. Or if it is the reference start vertex.
+    // NOTE: this check is necessary to handle the edge cases that may arise when a vertex has multiple exit paths but happens to lead to a vetex that needs a junction tree
+    private boolean vertexWarrantsJunctionTree(final MultiDeBruijnVertex vertex) {
+
+        for (MultiSampleEdge edge : outgoingEdgesOf(vertex)) {
+            if (inDegreeOf(getEdgeTarget(edge)) > 1) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    //TODO this is a placeholder for when these mechanisms should be more smart, as of right now I can live with the consequnces
```

Comment:
you're right here as well, I will unify this with its master. (When i was first extending and altering this class I copied and edited the methods as i needed but now that it is pretty clear what the changes tangibly are it shouldn't be too hard to avoid most of this duplication now)

Reply:
less than there was currently, i've pulled out these methods into into the interface as well.